commons/src/test/java/com/powsybl/commons/report/ReportNodeTest.java

```diff
@@ -320,21 +318,93 @@ void testTimestamps() {
+                .withMessageTemplate("unknown.key")
+                .build();
+        // translation should fall back to default properties as the key is not defined in the reports_en_US.properties
+        assertEquals("Cannot find message template with key: 'unknown.key'", report1.getMessage());
+
+        // With Locale.FRENCH
+        ReportNode report2 = ReportNode.newRootReportNode()
+                .withResourceBundles(PowsyblCoreTestReportResourceBundle.TEST_BASE_NAME, PowsyblCoreReportResourceBundle.BASE_NAME)
+                .withLocale(Locale.FRENCH)
+                .withMessageTemplate("unknown.key")
+                .build();
+        // translation should fall back to default properties as the key is not defined in the reports_en_US.properties
+        assertEquals("Template de message non trouvé pour la clé 'unknown.key'", report2.getMessage());
+    }
+
+    @Test
+    void testLocaleAndi18n() {
+        // Without giving a locale => default one is en_US
+        ReportNode rootReportEnglish = ReportNode.newRootReportNode()
+                .withResourceBundles(PowsyblCoreTestReportResourceBundle.TEST_BASE_NAME, PowsyblCoreReportResourceBundle.BASE_NAME)
+                .withMessageTemplateProvider(new BundleMessageTemplateProvider(TEST_BASE_NAME))
+                .withMessageTemplate("rootWithValue")
+                .withUntypedValue("value", 4)
+                .build();
+        // translation should fall back to default properties as the key is not defined in the reports_en_US.properties
+        assertEquals("Root message with value 4", rootReportEnglish.getMessage());
+        assertEquals(Locale.US, rootReportEnglish.getTreeContext().getLocale());
+
+        // With french locale
+        ReportNode rootReportFrench = ReportNode.newRootReportNode()
+                .withResourceBundles(TEST_BASE_NAME)
+                .withLocale(Locale.FRENCH)
+                .withMessageTemplate("rootWithValue")
+                .withUntypedValue("value", 4)
+                .build();
+        // translation should be from the reports_fr.properties file
+        assertEquals("Message racine avec la valeur 4", rootReportFrench.getMessage());
+        assertEquals(Locale.FRENCH, rootReportFrench.getTreeContext().getLocale());
+
+        // Test giving the specific France locale
+        ReportNode rootReportFrance = ReportNode.newRootReportNode()
+                .withResourceBundles(TEST_BASE_NAME)
+                .withLocale(Locale.FRANCE)
+                .withMessageTemplate("rootWithValue")
+                .withUntypedValue("value", 4)
+                .build();
+        // translation should be from the reports_fr.properties file as the key is not defined in the reports_fr_FR.properties
+        assertEquals("Message racine avec la valeur 4", rootReportFrance.getMessage());
+        assertEquals(Locale.FRANCE, rootReportFrance.getTreeContext().getLocale());
+    }
+
+    @Test
```

Comment:
```suggestion
    @Test
    void testMissingBundleName() {
        PowsyblException e = assertThrows(PowsyblException.class, () -> ReportNode.newRootReportNode().withResourceBundles().withMessageTemplate("unknown.key").build());
        assertEquals("bundleBaseNames must not be empty", e.getMessage());
    }
    
    @Test
```
Add test when missing bundle name